### ISSUE TITLE
Handlebars Caching causes possible error in PlainLauncher

### ISSUE THREAD

**Issue by marksreeves** [BorderTech/wcomponents#1157]
When testing `com.github.bordertech.wcomponents.lde.PlainLauncher` with the handlebars interceptor on MockRequest causes a SystemException as there is no Handlebars caching provider.

See `com.github.bordertech.wcomponents.lde.PlainLauncher_Test.testServer()`. Test this with the following addition to see the Exception:
``` java
Config.getInstance().setProperty(ConfigurationProperties.TEMPLATE_RENDERING_MODE, "on");
```

Rough chain (stack trace below) is:
* `com.github.bordertech.wcomponents.container.TemplateRenderInterceptor.paint(RenderContext)` calls (via several steps)
* `com.github.bordertech.wcomponents.template.HandlebarsCacheImpl.getCache()` calls (via several steps)
* `javax.cache.Caching.CachingProviderRegistry.getCachingProvider(ClassLoader)` which throws the exception.

Test output: 
```
SEVERE: Caught exception during render phase.
com.github.bordertech.wcomponents.util.SystemException: Problems with handlebars inline template. No CachingProviders have been configured
	at com.github.bordertech.wcomponents.template.HandlebarsRendererImpl.renderInline(HandlebarsRendererImpl.java:132)
	at com.github.bordertech.wcomponents.container.TemplateRenderInterceptor.paint(TemplateRenderInterceptor.java:83)
	at com.github.bordertech.wcomponents.container.AbstractContainerHelper.render(AbstractContainerHelper.java:252)
	at com.github.bordertech.wcomponents.servlet.ServletUtil.processRequest(ServletUtil.java:193)
	at com.github.bordertech.wcomponents.servlet.WServlet.serviceInt(WServlet.java:147)
	at com.github.bordertech.wcomponents.servlet.WServlet.service(WServlet.java:110)
	at com.github.bordertech.wcomponents.lde.TestServlet.service(TestServlet.java:341)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
	at com.github.bordertech.wcomponents.lde.TestServlet.service(TestServlet.java:129)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:503)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:557)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:154)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
	at org.eclipse.jetty.server.Server.handle(Server.java:370)
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:494)
	at org.eclipse.jetty.server.BlockingHttpConnection.handleRequest(BlockingHttpConnection.java:53)
	at org.eclipse.jetty.server.AbstractHttpConnection.headerComplete(AbstractHttpConnection.java:971)
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.headerComplete(AbstractHttpConnection.java:1033)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:644)
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)
	at org.eclipse.jetty.server.BlockingHttpConnection.handle(BlockingHttpConnection.java:72)
	at org.eclipse.jetty.server.bio.SocketConnector$ConnectorEndPoint.run(SocketConnector.java:264)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:608)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:543)
	at java.lang.Thread.run(Thread.java:745)
Caused by: javax.cache.CacheException: No CachingProviders have been configured
	at javax.cache.Caching$CachingProviderRegistry.getCachingProvider(Caching.java:381)
	at javax.cache.Caching$CachingProviderRegistry.getCachingProvider(Caching.java:351)
	at javax.cache.Caching.getCachingProvider(Caching.java:142)
	at javax.cache.Caching.getCache(Caching.java:279)
	at com.github.bordertech.wcomponents.template.HandlebarsCacheImpl.getCache(HandlebarsCacheImpl.java:71)
	at com.github.bordertech.wcomponents.template.HandlebarsCacheImpl.get(HandlebarsCacheImpl.java:50)
	at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:468)
	at com.github.jknack.handlebars.Handlebars.compileInline(Handlebars.java:438)
	at com.github.jknack.handlebars.Handlebars.compileInline(Handlebars.java:422)
	at com.github.bordertech.wcomponents.template.HandlebarsRendererImpl.renderInline(HandlebarsRendererImpl.java:123)
	... 32 more

Apr 04, 2017 6:39:39 PM com.github.bordertech.wcomponents.util.error.impl.DefaultSystemFailureMapper toMessage
SEVERE: The system is currently unavailable
com.github.bordertech.wcomponents.util.SystemException: Problems with handlebars inline template. No CachingProviders have been configured
	at com.github.bordertech.wcomponents.template.HandlebarsRendererImpl.renderInline(HandlebarsRendererImpl.java:132)
	at com.github.bordertech.wcomponents.container.TemplateRenderInterceptor.paint(TemplateRenderInterceptor.java:83)
	at com.github.bordertech.wcomponents.container.AbstractContainerHelper.render(AbstractContainerHelper.java:252)
	at com.github.bordertech.wcomponents.servlet.ServletUtil.processRequest(ServletUtil.java:193)
	at com.github.bordertech.wcomponents.servlet.WServlet.serviceInt(WServlet.java:147)
	at com.github.bordertech.wcomponents.servlet.WServlet.service(WServlet.java:110)
	at com.github.bordertech.wcomponents.lde.TestServlet.service(TestServlet.java:341)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
	at com.github.bordertech.wcomponents.lde.TestServlet.service(TestServlet.java:129)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:684)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:503)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:557)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:429)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:154)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
	at org.eclipse.jetty.server.Server.handle(Server.java:370)
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:494)
	at org.eclipse.jetty.server.BlockingHttpConnection.handleRequest(BlockingHttpConnection.java:53)
	at org.eclipse.jetty.server.AbstractHttpConnection.headerComplete(AbstractHttpConnection.java:971)
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.headerComplete(AbstractHttpConnection.java:1033)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:644)
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:235)
	at org.eclipse.jetty.server.BlockingHttpConnection.handle(BlockingHttpConnection.java:72)
	at org.eclipse.jetty.server.bio.SocketConnector$ConnectorEndPoint.run(SocketConnector.java:264)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:608)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:543)
	at java.lang.Thread.run(Thread.java:745)
Caused by: javax.cache.CacheException: No CachingProviders have been configured
	at javax.cache.Caching$CachingProviderRegistry.getCachingProvider(Caching.java:381)
	at javax.cache.Caching$CachingProviderRegistry.getCachingProvider(Caching.java:351)
	at javax.cache.Caching.getCachingProvider(Caching.java:142)
	at javax.cache.Caching.getCache(Caching.java:279)
	at com.github.bordertech.wcomponents.template.HandlebarsCacheImpl.getCache(HandlebarsCacheImpl.java:71)
	at com.github.bordertech.wcomponents.template.HandlebarsCacheImpl.get(HandlebarsCacheImpl.java:50)
	at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:468)
	at com.github.jknack.handlebars.Handlebars.compileInline(Handlebars.java:438)
	at com.github.jknack.handlebars.Handlebars.compileInline(Handlebars.java:422)
	at com.github.bordertech.wcomponents.template.HandlebarsRendererImpl.renderInline(HandlebarsRendererImpl.java:123)
	... 32 more
```

In addition, `com.github.bordertech.wcomponents.template.HandlebarsRendererImpl.getHandlebarsEngine(Map<String, Object>)` lines 177-183 are redundant:

``` java
if (isCaching()) {
	handlebars.with(CACHE);
}

if (isCaching()) {
	handlebars.with(CACHE);
}
```

**Comment by marksreeves** [comment]
In order to turn on the TemplateRenderInterceptor I had to modify the tests of PLainLauncher. See #1156. This is a **workaround** and not a solution to this bug.

**Comment by marksreeves** [comment]
Fixed in #1181
